lib/jquery-rjs/rendering.rb

```diff
@@ -1,13 +1,21 @@
 require 'action_view/helpers/rendering_helper'
 
-ActionView::Helpers::RenderingHelper.module_eval do
-  def render_with_update(options = {}, locals = {}, &block)
-    if options == :update
-      update_page(&block)
-    else
-      render_without_update(options, locals, &block)
-    end
+module JqueryRjs::RenderingHelper
+  method_name = "render#{'_with_update' if RUBY_VERSION < '2'}"
+  define_method(method_name) do |options = {}, locals = {}, &block|
+    update_page(&block) if options == :update
+
+    args = options, locals, block
+    RUBY_VERSION < '2' ? render_without_update(*args) : super(*args)
```

Comment:
> > but this is technically changing the existing functionality
>
> I don't see that. Where? (Note I had to fix some code with your first comment, should be the same semantics now though)

Sorry, that was unclear.  From the perspective of `input` to `output`, they should be the same.  I am more referring from evaluation stand point in which they differ (what if statements and other method calls they have to traverse to get said output, etc.).  I would explain further, but I think you get what I am saying (I hope).  Regardless, I was unclear... my bad.

> The issue with this is mostly the fact that I would have to either duplicate the entire module of helper methods (in the other patch in this PR) or do some Ruby module mincing of a third module before doing prepend/alias_method_chain. So...kind of a tossup. Thoughts on that?

I think it depends on if you buy into the B.S. argument I have been making this whole time :wink:.  That, and if it is worth the effort to do it this way.  That all said, isn't the only change that your code for the `selector_assertions.rb` be:

```ruby
module JqueryRjs::SelectorAssertions
  if RUBY_VERSION < '2'
    def response_from_page_with_rjs
      # ...
    end
  else
    def response_from_page
      # ...
    end
  end
end

module_to_patch = if defined?(ActionDispatch::Assertions::SelectorAssertions)
                    ActionDispatch::Assertions::SelectorAssertions
                  else
                    Rails::Dom::Testing::Assertions::SelectorAssertions
                  end

if RUBY_VERSION < '2'
  module_to_patch.module_eval do
    include JqueryRjs::SelectorAssertions
    alias_method_chain :response_from_page, :rjs
  end
else
  module_to_patch.prepend(JqueryRjs::SelectorAssertions)
end
```

Arguable, you could split out the code in `response_from_page_with_rjs` to be something like this to avoid duplication:

```ruby
def __response_from_page
  content_type = @response.content_type
 
   if content_type && Mime[:js] =~ content_type
    body = @response.body.dup
    root = HTML::Node.new(nil)

    while true
      next if body.sub!(RJS_STATEMENTS[:any]) do |match|
        html = unescape_rjs(match)
        matches = HTML::Document.new(html).root.children.select { |n| n.tag? }
        root.children.concat matches
        ""
      end
      break
    end
  
    root
  else
    yield
  end
end

if RUBY_VERSION < '2'
  def response_from_page_with_rjs
    __response_from_page { response_from_page_without_rjs }
  end
else
  def response_from_page
    __response_from_page { super() }
  end
end
```

But not sure if `super()` would work though and would be in the proper context.  The copy pasta way would be the safest probably.